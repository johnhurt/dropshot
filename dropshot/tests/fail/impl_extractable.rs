// Copyright 2021 Oxide Computer Company

// These "bad endpoint" tests are intended to throw compiler errors.  When a
// compiler error is thrown, imports which haven't yet successfully been used
// may throw "unused import" warnings. This is often a false positive, as
// the compiler error itself blocks intended usage.
//
// For this test (and other bad endpoint tests) disable the import warnings
// so output can be more precisely compared with the expected compiler error.
#![allow(unused_imports)]

/// This test verifies that outside of the crate, no types can be marked as
/// `Extractable`

use dropshot::Extractable;

struct SomethingCool;

impl Extractable for SomethingCool {}

fn main() {}
