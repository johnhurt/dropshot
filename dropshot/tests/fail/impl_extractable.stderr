error[E0277]: the trait bound `SomethingCool: dropshot::handler::private::Seal` is not satisfied
   --> $DIR/impl_extractable.rs:19:6
    |
19  | impl Extractable for SomethingCool {}
    |      ^^^^^^^^^^^ the trait `dropshot::handler::private::Seal` is not implemented for `SomethingCool`
    |
   ::: $WORKSPACE/dropshot/src/handler.rs
    |
    | pub trait Extractable: private::Seal {}
    |                        ------------- required by this bound in `Extractable`
